core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1878,25 +1879,25 @@ private void journalSetAttribute(long fileId, long opTimeMs, SetAttributeOptions
    */
   public void scheduleAsyncPersistence(AlluxioURI path) throws AlluxioException {
     synchronized (mInodeTree) {
-      scheduleAsyncPersistenceInternal(path);
       long fileId = mInodeTree.getInodeByPath(path).getId();
+      scheduleAsyncPersistenceInternal(fileId);
       // write to journal
       AsyncPersistRequestEntry asyncPersistRequestEntry =
           AsyncPersistRequestEntry.newBuilder().setFileId(fileId).build();
       writeJournalEntry(
           JournalEntry.newBuilder().setAsyncPersistRequest(asyncPersistRequestEntry).build());
       flushJournal();
     }
+    mAsyncPersistHandler.scheduleAsyncPersistence(path);
```

Comment:
when moved outside the critical session, the `path` could become invalid (e.g., renamed or removed) when reaching here. Is it still correct here?